src/Uno.UI/UI/Xaml/ResourceBindingCollection.cs

```diff
@@ -16,7 +16,25 @@ internal class ResourceBindingCollection
 	private readonly Dictionary<DependencyProperty, Dictionary<DependencyPropertyValuePrecedences, ResourceBinding>> _bindings = new();
 	private BindingEntry[]? _cachedAllBindings;
 
-	public bool HasBindings => _bindings.Count > 0 && _bindings.Any(b => b.Value.Any());
+	public bool HasBindings
+	{
+		get
+		{
+			// Avoiding the use of LINQ here to reduce unnecessary enumerator boxing.
+			if (_bindings.Count > 0)
+			{
+				foreach (var (_, value) in _bindings)
```

Comment:
While nice to write, it still call deconstruct and it probably would be interesting to just do `var entry`, then do `entry.Value.Count` after.

See in [sharplab](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwCwG5nJQDMmM6AwugN7Lp3q30AOATgJYBuAhgC4Cm6ACJsAxjzYB7AHZcWATwA8bKTwA0Q0eOmzFytej0A+QwH50AfQBGy4MoDmAZ3xJ6DF/WLpLEiQBt0ALIAFACUbq407q50jNEGAGboQVY29g4AdOQSAK4q6IboqCHIAJBlkWVl8RIsfFwiABZJ3CxJ5urcvtl8YcoW1lK2Uo7FSJUlFeNsiUGd3Zk5eQVFpeOT41AA7Og8LN3O4yUAvqtlJ2NnBBclW+jxXL4OfM5xMVHo5+eEJLUiNcDoBy7bJidAAIVSwwAoip5EE9OgAAosCRMPgsHhydQIiGDewhXBAA).